### PR TITLE
Feature/exercises service

### PR DESCRIPTION
**Pull request by Hairax**
# US 2 Loading exercise lists for routines

## Description

The connection was created with the external service that works as our exercise bank, along with the key points of all the exercises and exercises by name.

## Test notes

To test the implementation you can use `Postman` or `RapidAPI Client` and use the next directions:

- `GET` - `http://localhost:4444/exercises/`
- `GET` - `http://localhost:4444/exercises/"name"`
example 
- `GET` - `http://localhost:4444/exercises/bike`

For ones that need body you can use the next one:

```json
[
  {
    "bodyPart": "waist",
    "equipment": "body weight",
    "gifUrl": "https://v2.exercisedb.io/image/L2q3TJWcp8iNRb",
    "id": "0003",
    "name": "air bike",
    "target": "abs",
    "secondaryMuscles": [
      "hip flexors"
    ],
    "instructions": [
      "Lie flat on your back with your hands placed behind your head.",
      "Lift your legs off the ground and bend your knees at a 90-degree angle.",
      "Bring your right elbow towards your left knee while simultaneously straightening your right leg.",
      "Return to the starting position and repeat the movement on the opposite side, bringing your left elbow towards your right knee while straightening your left leg.",
      "Continue alternating sides in a pedaling motion for the desired number of repetitions."
    ]
  }, ...
```

## Links

- [ClickUp Ticket](https://app.clickup.com/t/868abwd5j)

### CODE REVIEW

**Comment by Miguelito2774** [comment]
We should hide our API token in the repo secrets, also solve your merge conflcts.

File: apps/engage-hub-service/src/ExerciseService/exercise.controller.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  @MessagePattern({ cmd: 'filterExercisesByName' })
+  async filterExercisesByName({ name }: { name: string }): Promise<any> {
+    const exercises = await this.exerciseService.getExercises();
+
+    // Filtra según la estructura de datos correcta
+    const filteredExercises = Array.isArray(exercises)
+      ? exercises.filter((exercise) =>
+          exercise.name.toLowerCase().includes(name.toLowerCase())
+        )
+      : []; // Asegura que sea un array para filtrar correctamente
```
Comment (AndrePzDn) [inline apps/engage-hub-service/src/ExerciseService/exercise.controller.ts]: This should be done in a service and not in the controller that get filtered exercises

File: infraestructure/api-gateway/src/app.controller.ts [hunk 2]
```diff
@@ -37,4 +37,14 @@ export class AppController {
...
   }
+
+  @Get('engage/exercises')
+  getExercises(): Observable<any> {
+    return this.engageHubService.send({ cmd: 'getExercises' }, {});
+  }
+
+  @Get('engage/exercises/filterByName')
+  filterExercises(@Query('name') name: string): Observable<any> {
+    return this.engageHubService.send({ cmd: 'filterExercisesByName' }, { name });
```
Comment (AndrePzDn) [inline infraestructure/api-gateway/src/app.controller.ts]: You should separate endpoint in different controllers not all in `app.controller.ts`

**Review by AndrePzDn — CHANGES_REQUESTED** [review]

File: infraestructure/api-gateway/src/app.controller.ts [hunk 3]
```diff
@@ -37,4 +37,17 @@ export class AppController {
...
+    return this.engageHubService.send({ cmd: 'getExercises' }, {});
+  }
+
+  @Get('engage/exercises/filterByName')
+  filterExercises(@Query('name') name: string): Observable<any> {
+    return this.engageHubService.send(
+      { cmd: 'filterExercisesByName' },
+      { name },
+    );
+  }
```
Comment (AndrePzDn) [inline infraestructure/api-gateway/src/app.controller.ts]: Duplicated code it already have its own controller

```suggestion
```

File: infraestructure/api-gateway/src/exercises/exercises.controller.ts
Comment (AndrePzDn) [inline infraestructure/api-gateway/src/exercises/exercises.controller.ts:1]: Add endline pls

**Review by AndrePzDn — CHANGES_REQUESTED** [review]

**Review by Miguelito2774 — APPROVED** [review]
good job!

**Review by AndrePzDn — APPROVED** [review]

**Pull request merged**